cpp/src/transform/compute_column.cu

```diff
@@ -117,17 +121,32 @@ std::unique_ptr<column> compute_column(table_view const& table,
   auto const config          = cudf::detail::grid_1d{table.num_rows(), block_size};
   auto const shmem_per_block = parser.shmem_per_thread * config.num_threads_per_block;
 
-  // Execute the kernel
   auto table_device = table_device_view::create(table, stream);
-  if (has_nulls) {
-    cudf::detail::compute_column_kernel<MAX_BLOCK_SIZE, true>
+  // Use template metaprogramming to select the appropriate kernel based on has_nulls and
+  // has_complex_type
+  auto launch_kernel = [&](auto has_nulls_tag, auto has_complex_type_tag) {
+    using HasNulls       = decltype(has_nulls_tag);
+    using HasComplexType = decltype(has_complex_type_tag);
+
+    cudf::detail::compute_column_kernel<MAX_BLOCK_SIZE, HasNulls::value, HasComplexType::value>
       <<<config.num_blocks, config.num_threads_per_block, shmem_per_block, stream.value()>>>(
         *table_device, device_expression_data, *mutable_output_device);
+  };
```

Comment:
Since there are 4 copies of this kernel per the combination of bool templates, perhaps each of these could be defined in separate .cu files to help reduce compile time. There should be no effect on runtime performance.

Reply:
Done. I was hoping to sneak the code in as-is and was praying David wouldn’t catch the issue. I apologize for even having that thought in the first place. :sweat_smile: